source/kraken/fill_disruption_from_chaos.cpp

```diff
@@ -211,27 +217,240 @@ make_impact(const chaos::Impact& chaos_impact, nt::PT_Data& pt_data) {
     return std::move(impact);
 }
 
-void delete_disruption(nt::PT_Data& pt_data,
-                    const std::string& disruption_id) {
+struct apply_impact_visitor : public boost::static_visitor<> {
+    boost::shared_ptr<nt::new_disruption::Impact> impact;
+    nt::PT_Data& pt_data;
+    std::function<bool(nt::VehicleJourney&)> f;
+
+    apply_impact_visitor(boost::shared_ptr<nt::new_disruption::Impact> impact,
+            nt::PT_Data& pt_data, std::function<bool(nt::VehicleJourney&)> f) :
+     impact(impact), pt_data(pt_data), f(f){
+     }
+
+    void operator()(nt::new_disruption::UnknownPtObj&) {
+    }
+
+    void operator()(const nt::Network * network) {
+        for(auto line : network->line_list) {
+            this->operator()(line);
+        }
+    }
+    void operator()(const nt::StopArea * ) {
+    }
+    void operator()(nt::new_disruption::LineSection & ls) {
+        this->operator()(ls.line);
+    }
+    void operator()(const nt::Line *line) {
+        for(auto route : line->route_list) {
+            this->operator()(route);
+        }
+    }
+    void operator()(const nt::Route * route) {
+        for (auto journey_pattern : route->journey_pattern_list) {
+            journey_pattern->for_each_vehicle_journey(f);
+        }
+    }
+};
+
+void apply_impact(boost::shared_ptr<nt::new_disruption::Impact>impact,
+        nt::PT_Data& pt_data, nt::MetaData& meta) {
+    if (impact->severity->effect != nt::new_disruption::Effect::NO_SERVICE) {
+        return;
+    }
+    auto f = [&](nt::VehicleJourney& vj) {
```

Comment:
hum not sure to be fan of the functor passed to handle either the add or the deletion of a disruption.

A more static approach might be more suitable, we can discuss it tomorrow if you want.